src/main/kotlin/org/rust/lang/core/types/RustTypificationEngine.kt

```diff
@@ -235,3 +245,54 @@ private fun RsImplItem.remapTypeParameters(
                 null
             }
         }.toMap()
+
+
+fun RsExpr.isConvertibleTo(type: RustType): Boolean = type.findSubtypes(project).contains(type)
+
+fun RustType.findSubtypes(project: Project): Sequence<RustType> {
+    val subtypes = mutableSetOf<RustType>()
+    findSubtypes(subtypes, project)
+    return subtypes.asSequence()
+}
+
+private fun RustType.findSubtypes(subtypes: MutableSet<RustType>, project: Project) {
+    if (!subtypes.add(this)) {
+        return
+    }
+    if (this is RustReferenceType) {
+        RustPointerType(referenced, mutable).findSubtypes(subtypes, project)
+        if (mutable) {
+            RustReferenceType(referenced, false).findSubtypes(subtypes, project)
+        }
+
+        val base = stripAllRefsIfAny()
+        base.getImplsIn(project, "Deref")
+            .forEach { impl: RsImplItem ->
+                impl.findAssocType("Target")
+                    ?.typeReference?.type?.findSubtypes(subtypes, project)
```

Comment:
Do we have a test that `A: Deref<Target=B>` and `B: Deref<Target=A>`. I see this should be handled by the check at the start of the function, but let's add a test nevertheless.